modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/helper/search/BookmarkSearchHandler.kt

```diff
@@ -0,0 +1,37 @@
+package au.com.shiftyjelly.pocketcasts.podcasts.helper.search
+
+import au.com.shiftyjelly.pocketcasts.models.entity.Bookmark
+import au.com.shiftyjelly.pocketcasts.repositories.bookmark.BookmarkManager
+import io.reactivex.Observable
+import io.reactivex.Single
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class BookmarkSearchHandler @Inject constructor(
+    private val bookmarkManager: BookmarkManager,
+) : SearchHandler<Bookmark>() {
+    override val searchDebounce = 500L
+
+    override fun getSearchResultsObservable(podcastUuid: String): Observable<SearchResult> =
+        searchQueryRelay.debounce { // Only debounce when search has a value otherwise it slows down loading the pages
```

Comment:
I don't think we need to include a debounce as we aren't calling off to a server.

Reply:
Removed in 0bd594e